pkg/cloud/gcp/actuators/machine/machine_scope_test.go

```diff
@@ -0,0 +1,450 @@
+						Name:      "test",
+						Namespace: defaultNamespaceName,
+						Labels: map[string]string{
+							machinev1.MachineClusterIDLabel: "CLUSTERID",
+						},
+					},
+					Spec: machinev1.MachineSpec{
+						ProviderSpec: machinev1.ProviderSpec{
+							Value: validProviderSpec,
+						},
+					}},
+			},
+			expectedError: errors.New("error getting credentials secret \"credentials-test\" in namespace \"test\": secrets \"credentials-test\" not found"),
+		},
+		{
+			name: "fail to getting project from json key",
+			params: machineScopeParams{
+				coreClient:           controllerfake.NewFakeClient(userDataSecret, invalidCredentialsSecret),
+				computeClientBuilder: computeservice.MockBuilderFuncType,
+				machine: &machinev1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: defaultNamespaceName,
+						Labels: map[string]string{
+							machinev1.MachineClusterIDLabel: "CLUSTERID",
+						},
+					},
+					Spec: machinev1.MachineSpec{
+						ProviderSpec: machinev1.ProviderSpec{
+							Value: validProviderSpec,
+						},
+					}},
+			},
+			expectedError: errors.New(`error getting project from JSON key: error un marshalling JSON key: json: cannot unmarshal number into Go value of type struct { ProjectID string "json:\"project_id\"" }`),
+		},
+		{
+			name: "fail to create compute service",
+			params: machineScopeParams{
+				coreClient: fakeClient,
+				computeClientBuilder: func(serviceAccountJSON string) (computeservice.GCPComputeService, error) {
+					return nil, errors.New("test error")
+				},
+				machine: &machinev1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: defaultNamespaceName,
+						Labels: map[string]string{
+							machinev1.MachineClusterIDLabel: "CLUSTERID",
+						},
+					},
+					Spec: machinev1.MachineSpec{
+						ProviderSpec: machinev1.ProviderSpec{
+							Value: validProviderSpec,
+						},
+					}},
+			},
+			expectedError: errors.New("error creating compute service: test error"),
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			scope, err := newMachineScope(tc.params)
+
+			if tc.expectedError != nil {
+				if err == nil {
+					t.Fatal("newMachineScope() was expected to return error")
+				}
+				if err.Error() != tc.expectedError.Error() {
+					t.Fatalf("expected: %v, got %v", tc.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("newMachineScope() was not expected to return error: %v", err)
+				}
+				if scope.Context != context.Background() {
+					t.Fatal("expected to machine scope to be context.Background()")
+				}
+
+				expectedProviderID := "gce://test//test"
+				if scope.providerID != expectedProviderID {
+					t.Fatalf("expected to provider id to be %s, got: %s", expectedProviderID, scope.providerID)
+				}
+			}
+		})
+	}
+}
+
+func TestPatchMachine(t *testing.T) {
+	os.Setenv("TEST_ASSET_KUBE_APISERVER", "/tmp/kubebuilder/bin/kube-apiserver")
```

Comment:
why are hardcoding this here?
If overriding defaults were to be needed this should be handled in the makefile.

Reply:
It's a leftover after I debugged the coded inside IDE